crates/pico/src/derived_node.rs

```diff
@@ -1,29 +1,43 @@
-use std::{fmt, hash::Hash};
+use std::{fmt, hash::Hash, marker::PhantomData, ops::Deref};
+
+use intern::{intern_struct, InternId};
+use serde::{Deserialize, Serialize};
+use tinyvec::ArrayVec;
```

Comment:
should we use `TinyVec`? I mean, maybe we will never have >8 params, but it would be nice to be able to do something about that.

Or maybe the macro can intern params 8+ as a tuple, and you just accept some perf loss in that case

Reply:
Let's intern a tuple in that case. When we integrate it to the compiler, I think, it would be easier to measure the impact of having tinyvec here or something else.

Reply:
I don't consider that a blocker for merging to master though fwiw

Reply:
I'll do it as a follow up PR with additional test.